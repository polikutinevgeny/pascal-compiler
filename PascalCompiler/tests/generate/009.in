var a, b: Integer;
begin
    b := 2;
    a := 5;
    a := a * b + b div a;
    write(a);
end.