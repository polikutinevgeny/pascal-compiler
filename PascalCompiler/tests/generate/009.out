.686p
include \masm32\include\masm32rt.inc
.mmx
.xmm
.code
main PROC
Push Ebp
Mov Ebp, Esp
Sub Esp, 8
Push 2
Pop Eax
Mov dword ptr [Ebp-8], Eax
Push 5
Pop Eax
Mov dword ptr [Ebp-4], Eax
Push dword ptr [Ebp-4]
Push dword ptr [Ebp-8]
Pop Ebx
Pop Eax
Imul Ebx
Push Eax
Push dword ptr [Ebp-8]
Push dword ptr [Ebp-4]
Pop Ebx
Pop Eax
Cdq
Idiv Ebx
Push Eax
Pop Ebx
Pop Eax
Add Eax, Ebx
Push Eax
Pop Eax
Mov dword ptr [Ebp-4], Eax
Push dword ptr [Ebp-4]
Mov Eax, Esp
printf("%d", dword ptr [Eax])
Sub Esp, 4
Mov Esp, Ebp
Pop Ebp
exit
main ENDP
END main
